Source/ACE/Entity/Player.cs

```diff
@@ -1106,39 +1126,112 @@ public void HandleActionApproachVendor(Vendor vendor, List<WorldObject> itemsFor
         /// <param name="items"></param>
         public void HandleActionBuy(ObjectGuid vendorId, List<ItemProfile> items)
         {
-            ActionChain chain = new ActionChain();
-            CurrentLandblock.ChainOnObject(chain, vendorId, (WorldObject vdr) =>
-            {
-                (vdr as Vendor).BuyItemsStartTransaction(vendorId, items, this);
-            });
-            chain.EnqueueChain();
-        }
-
-        public void HandleActionBuyStartTransaction(WorldObject vendor, List<WorldObject> purchaselist, uint cost)
-        {
             new ActionChain(this, () =>
             {
-                // todo: check inventoy space.
+                List<ItemProfile> filteredlist = new List<ItemProfile>();
```

Comment:
This is just too much code in a Lambda, and it's simply in the wrong class.  Make this a function in the Vendor class - it doesn't belong in player.